core/src/main/java/com/linecorp/armeria/client/HttpResponseDecoder.java

```diff
@@ -121,13 +122,13 @@ final boolean reserveUnfinishedResponse(int maxUnfinishedResponses) {
     }
 
     final void failUnfinishedResponses(Throwable cause) {
-        try {
-            for (HttpResponseWrapper res : responses.values()) {
-                res.close(cause);
-            }
-        } finally {
-            unfinishedResponses -= responses.size();
-            responses.clear();
+        for (final Iterator<HttpResponseWrapper> iterator = responses.values().iterator();
+             iterator.hasNext();) {
+            final HttpResponseWrapper res = iterator.next();
+            // To avoid calling removeResponse by res.close(cause), remove before closing.
+            iterator.remove();
```

Comment:
or we should just copy the collection & and call close only without decreasing `unfinishedResponses` too? If `res#close` is intended to call `removeResponse`.

Reply:
> we should just copy the collection

I think we don't have to. 😄 
Would it be possible to make a test for this?
I'm fine without the test though. 😄